### PR TITLE
linalg: `hermitian`

### PR DESCRIPTION
**Pull request by perazz**
Address #883: compute the Hermitian version of a rank-2 matrix. 
I agree that having a `hermitian` helper function is inline with stdlib already offering several others, including `is_hermitian`.

Fortran uses verbs (`transpose`), so there is no direct equivalent. I believe another possible name for this function may be just `hermite(A)`, but probably `hermitian` is more in sync with `is_hermitian`.

### Proposed implementation

- `A_h = hermitian(A)`

### Key facts
- For `complex` matrices, the function returns the conjugate transpose (`conjg(transpose(a))`). 
- For `real` or `integer` matrices, the function returns the transpose (`transpose(a)`).

### Prior art
- SciPy: `A.H` (`H` operator)
- NumPy: `A.H` (`H` operator)

cc: @jvdp1 @jalvesz @loiseaujc

### CODE REVIEW

File: doc/specs/stdlib_linalg.md [hunk 1]
```diff
@@ -509,6 +509,37 @@ Returns a `logical` scalar that is `.true.` if the input matrix is skew-symmetri
...
+`h = ` [[stdlib_linalg(module):hermitian(interface)]] `(a)`
+
+### Arguments
+
+`a`: Shall be a rank-2 array of type `integer`, `real`, or `complex`. The input matrix `a` is not modified.
+
+### Return value
+
+Returns a rank-2 array of the same shape and type as `a`. If `a` is of type `complex`, the Hermitian matrix is computed as `conjg(transpose(a))`. 
+For `real` or `integer` types, it is equivalent to `transpose(a)`.
```
Comment (jvdp1) [inline doc/specs/stdlib_linalg.md:535]:
```suggestion
For `real` or `integer` types, it is equivalent to the intrinsic `transpose(a)`.
```

File: src/stdlib_linalg.fypp [hunk 2]
```diff
@@ -1546,4 +1583,5 @@ contains
       end function is_hessenberg_${t1[0]}$${k1}$
     #:endfor
     
+
```
Comment (jvdp1) [inline src/stdlib_linalg.fypp:1586]:
```suggestion
```

**Review by jvdp1 — APPROVED** [review]
Thank you @perazz . Nice and clean implementation.

**Review by jalvesz — APPROVED** [review]

**Comment by perazz** [comment]
Thank you @jvdp1 @jalvesz, I will merge this as there have been no further comments.

**Pull request merged**